share/getters/shrex.go

```diff
@@ -136,6 +136,10 @@ func (sg *ShrexGetter) GetEDS(ctx context.Context, header *header.ExtendedHeader
 	}()
 
 	dah := header.DAH
+	// short circuit if the data root is empty
+	if dah.Equals(share.EmptyRoot()) {
+		return share.EmptyExtendedDataSquare(), nil
+	}
```

Comment:
should we just have this first? Does it even make sense to start the span for this?

Reply:
yes